src/lm.jl

```diff
@@ -123,31 +123,38 @@ const FIT_LM_DOC = """
 
 """
     fit(LinearModel, formula, data, allowrankdeficient=false;
-       [wts::AbstractVector], dropcollinear::Bool=true)
+       [wts::FrequencyWeights], dropcollinear::Bool=true)
     fit(LinearModel, X::AbstractMatrix, y::AbstractVector;
-        wts::AbstractVector=similar(y, 0), dropcollinear::Bool=true)
+        wts::FrequencyWeights=fweights(Float64[]), dropcollinear::Bool=true)
```

Comment:
Shouldn't this be

```suggestion
        wts::FrequencyWeights=fweights(similar(y, 0)), dropcollinear::Bool=true)
```

to keep it consistent with the current implementation? Hardcoding `Float64` would mean that there's likely unintended promotion to `Float64` for `Float32`-user input